### PR TITLE
Adding files

### PR DESCRIPTION
**Pull request by YuKuroA**

### CODE REVIEW

File: task-board/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import { useState } from "react";
+
+
+function App() {
+  const [myLogin, setMyLogin] = useState('')
+
+  const passLogin = (login:string) => {
+    setMyLogin(login)
+  }
```
Comment (IlIAntonIlI) [inline task-board/src/App.tsx]: add enter before return

File: task-board/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import './App.css';
+import 'typeface-inter';
+import UserBoard from './pages/userBoard/userBoard';
+import SignInPage from './pages/signIn/signInPage';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import { useState } from "react";
+
+
+function App() {
+  const [myLogin, setMyLogin] = useState('')
```
Comment (IlIAntonIlI) [inline task-board/src/App.tsx]: missed semi-colon at the end

File: task-board/src/App.tsx (hunk 1 above)
Comment (IlIAntonIlI) [reply task-board/src/App.tsx]: missed semi-colon at the end of line should be `setMyLogin(login);`

File: task-board/src/components/addColumn/addColumn.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        event.preventDefault();
+
+        const target = event.target as typeof event.target & {
+            columnName: { value: string };
+        };
+
+        addColumn({
+            id: crypto.randomUUID(),
+            columnName: target.columnName.value
+        })
```
Comment (IlIAntonIlI) [inline task-board/src/components/addColumn/addColumn.tsx]: semi-colon

File: task-board/src/components/addColumn/addColumn.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        closeModal()
```
Comment (IlIAntonIlI) [inline task-board/src/components/addColumn/addColumn.tsx]: semi-colon

File: task-board/src/components/addColumn/addColumn.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        <form className="addColumnForm" onSubmit={createColumn}>
+            <input type="text" placeholder="Enter column title" name="columnName" />
+            <div className="controlsAddColumn">
+                <button type="submit">Add column</button>
+                <img src={abandonIcon} alt="abadon" onClick={closeModal} />
+            </div>
+        </form>
+    )
+
+    return addColumnForm
```
Comment (IlIAntonIlI) [inline task-board/src/components/addColumn/addColumn.tsx]: semi-colon

File: task-board/src/components/addColumn/addColumn.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+export default AddColumn
```
Comment (IlIAntonIlI) [inline task-board/src/components/addColumn/addColumn.tsx]: semi-colon

Comment (IlIAntonIlI) [reply task-board/src/components/addColumn/addColumn.tsx]: review all files, i see missed semi-colons below as well

File: task-board/src/components/card/card.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                <p>{props.cardName}</p>
+                <img src={deleteIcon} alt="delete button" onClick={props.onDelete} />
+            </div>}
+            {showForm &&
+            <form onSubmit={editMyCard}>
+                <input type="text" name="cardName" />
+                <button type="submit">Submit</button>
+            </form>}
+        </div>
+
```
Comment (IlIAntonIlI) [inline task-board/src/components/card/card.tsx]: remove unnecessary whiteline
Also i would suggest to make code above looks like:
```
 <div>
    {
      !showForm &&
        <div className="card" onClick={callEdit}>
            <p>{props.cardName}</p>
            <img src={deleteIcon} alt="delete button" onClick={props.onDelete} />
        </div>
    }
    {
      showForm &&
        <form onSubmit={editMyCard}>
            <input type="text" name="cardName" />
            <button type="submit">Submit</button>
        </form>
    }
</div>
```

File: task-board/src/components/addColumn/addColumn.css [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    font-style: normal;
+    font-size: 20px;
+    line-height: 24px;
+    letter-spacing: 0.5px;
+}
+
+.controlsAddColumn img {
+    width: 20px;
+    height: 20px;
+}
```
Comment (IlIAntonIlI) [inline task-board/src/components/addColumn/addColumn.css]: one more small comment, all files should have an empty line in the end

File: task-board/src/components/card/card.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+interface CardProps {
+    id: string;
+    cardName: string;
+    onDelete: () => void;
+    editCard: (newCard: MyCard) => void;
+}
+
+const Card: React.FC<CardProps> = (props) => {
+    const [showForm, setShowForm] = useState(false)
```
Comment (IlIAntonIlI) [inline task-board/src/components/card/card.tsx]: I guess this variable applies to editCardForm, so its would be more clear to name it with more specific name it like `showEditCardForm, setShowEditCardForm`

File: task-board/src/components/card/card.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+    const callEdit = () => {
+        setShowForm(true)
+    }
+
+    const editMyCard = (event : React.FormEvent<HTMLFormElement>) => {
```
Comment (IlIAntonIlI) [inline task-board/src/components/card/card.tsx]: remove space aftervariable name, should be `(event: React.FormEvent<HTMLFormElement>)`

Comment (IlIAntonIlI) [reply task-board/src/components/card/card.tsx]: review other places too, please

File: task-board/src/App.tsx (hunk 2 above)
Comment (IlIAntonIlI) [reply task-board/src/App.tsx]: also I would suggest to use smth like `login`, `username` or `userLogin` as variable name

File: task-board/src/components/signIn/signIn.tsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+        const target = event.target as typeof event.target & {
+            login: { value: string };
+            password: { value: string };
+        };
+
+        const login = target.login.value;
+        const password = target.password.value;
+
+        passLogin(login)
```
Comment (IlIAntonIlI) [inline task-board/src/components/signIn/signIn.tsx]: do you set a login of user, before he input correct info? :D

Should called after username and password passed validation

File: task-board/src/components/signIn/signIn.tsx [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+        setIsLoginCorrect(false);
+
+        if (users.find(user => user.login==login)) {
```
Comment (IlIAntonIlI) [inline task-board/src/components/signIn/signIn.tsx]: spaces, should be `if (users.find(user => user.login === login)) {`, also dont use `==`, its should be `===` in most cases

Comment (IlIAntonIlI) [reply task-board/src/components/signIn/signIn.tsx]: also replace it everywhere pls

File: task-board/src/components/taskBoard/taskBoard.tsx [hunk 13]
```diff
@@ -11,7 +11,7 @@ const firstColumn:MyColumn = ({
 
 const TaskBoard:React.FC = () => {
     const [showForm, setShowForm] = useState(false);
-    const [columnList, setColumnList] = useState([firstColumn]);
+    const [columnList, setColumnList] = useState ([firstColumn]);
```
Comment (IlIAntonIlI) [inline task-board/src/components/taskBoard/taskBoard.tsx]: remove space after useState, should be `const [columnList, setColumnList] = useState([firstColumn]);`

File: task-board/src/pages/userBoard/userBoard.tsx [hunk 14]
```diff
@@ -2,18 +2,22 @@ import Header from "../../components/header/header"
 import TaskBoard from "../../components/taskBoard/taskBoard"
 import './userBoard.css'
 
+
```
Comment (IlIAntonIlI) [inline task-board/src/pages/userBoard/userBoard.tsx]: unnecessary whiteline

File: task-board/src/pages/userBoard/userBoard.tsx [hunk 15]
```diff
@@ -2,18 +2,22 @@ import Header from "../../components/header/header"
...
 function getUsers() {
     // TODO: Add fetching data about users
     return [];
 }
 
+
```
Comment (IlIAntonIlI) [inline task-board/src/pages/userBoard/userBoard.tsx]: unnecessary whiteline

File: task-board/src/pages/userBoard/userBoard.tsx [hunk 16]
```diff
@@ -2,18 +2,22 @@ import Header from "../../components/header/header"
...
 interface UserBoardProps {
     login: string
 }
 
 const UserBoard: React.FC<UserBoardProps> = ({login}) => {
+
```
Comment (IlIAntonIlI) [inline task-board/src/pages/userBoard/userBoard.tsx]: unnecessary whiteline

File: task-board/src/pages/userBoard/userBoard.tsx [hunk 17]
```diff
@@ -2,18 +2,22 @@ import Header from "../../components/header/header"
...
     const users = getUsers();    
 
+
```
Comment (IlIAntonIlI) [inline task-board/src/pages/userBoard/userBoard.tsx]: unnecessary whiteline

File: task-board/src/pages/userBoard/userBoard.tsx [hunk 18]
```diff
@@ -2,18 +2,22 @@ import Header from "../../components/header/header"
...
     // TODO: Add fetching data about users
     return [];
 }
 
+
 interface UserBoardProps {
     login: string
 }
 
 const UserBoard: React.FC<UserBoardProps> = ({login}) => {
```
Comment (IlIAntonIlI) [inline task-board/src/pages/userBoard/userBoard.tsx]: all variables and parameters inside {}, should be written with spaces
`({ login })`, 
```
{ authData: someData, user: User, ...destructedObject }
```

Doesn't applies to markup part: 
`{!isPasswordCorrect && <p className='error-message'>Password is not correct</p>}`

File: task-board/src/App.tsx (hunk 2 above)
Comment (YuKuroA) [reply task-board/src/App.tsx]: 1. Added semi-colons at the end of lines 10, 13, 14;
2. Changed variable name from "myLogin" to "userLogin" (lines: 10, 19);
3. Changed function name from "setMyLogin" to "setUserLogin" (lines: 10, 13);

File: task-board/src/App.tsx (hunk 1 above)
Comment (YuKuroA) [reply task-board/src/App.tsx]: 1. Added enter before return (line 15);
2. Added semi-colons at the end of lines 10, 13, 14;

File: task-board/src/components/addColumn/addColumn.css (hunk 8 above)
Comment (YuKuroA) [reply task-board/src/components/addColumn/addColumn.css]: Added line in the end to this file. 
Checking another

File: task-board/src/components/addColumn/addColumn.tsx (hunk 3 above)
Comment (YuKuroA) [reply task-board/src/components/addColumn/addColumn.tsx]: 1. Added semi-colon to lines: 8, 9, 23, 25, 38, 42

File: task-board/src/components/addColumn/addColumn.tsx (hunk 4 above)
Comment (YuKuroA) [reply task-board/src/components/addColumn/addColumn.tsx]: Resolved

File: task-board/src/components/addColumn/addColumn.tsx (hunk 5 above)
Comment (YuKuroA) [reply task-board/src/components/addColumn/addColumn.tsx]: Resolved

File: task-board/src/components/addColumn/addColumn.tsx (hunk 6 above)
Comment (YuKuroA) [reply task-board/src/components/addColumn/addColumn.tsx]: 1. Added semi-colon
2. Added line at the end of file

File: task-board/src/components/card/card.css [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 19px;
+    color: #000000;
+    margin-top: 8px;
+    margin-bottom: 8px;
+    margin-left: 12px;
+    margin-right: 12px;
+    word-wrap: break-word;
+}
```
Comment (YuKuroA) [inline task-board/src/components/card/card.css]: 1. Added line at the end of the file

File: task-board/src/components/card/card.tsx (hunk 9 above)
Comment (YuKuroA) [reply task-board/src/components/card/card.tsx]: 1. Changed variable name from "showForm" to "showEditCardForm".
2. Changed function name from "setShowForm" to "setShowEditCardForm"

File: task-board/src/components/addColumn/addColumn.tsx [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+import 'typeface-inter';
+import { MyColumn } from "../../types/columntype";
+
+interface FormProps {
+    addColumn: (newColumn: MyColumn) => void
+    closeModal: () => void
+}
+
+const AddColumn: React.FC<FormProps> = ({addColumn, closeModal}) => {
+    const createColumn = (event : React.FormEvent<HTMLFormElement>) => {
```
Comment (YuKuroA) [inline task-board/src/components/addColumn/addColumn.tsx]: 1. Removed spase from (event : React.FormEvent<HTMLFormElement>)

File: task-board/src/components/card/card.tsx (hunk 10 above)
Comment (YuKuroA) [reply task-board/src/components/card/card.tsx]: 1. Removed space in ` (event : React.FormEvent<HTMLFormElement>)`  in this file
2. Removed space in ` (event : React.FormEvent<HTMLFormElement>)`  in file "addCard.tsx"
3. Removed space in ` (event : React.FormEvent<HTMLFormElement>)`  in file "addColumn.tsx"

File: task-board/src/components/card/card.tsx (hunk 7 above)
Comment (YuKuroA) [reply task-board/src/components/card/card.tsx]: 1. Removed whiteline
2. Added semi-colons for lines: 16, 19, 32, 34, 56, 59
3. Made code from line 39 to 54 prettier
4. Added line at the end of the file

File: task-board/src/components/header/header.css [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    
+    .login button {
+        display: block;
+    }
+
+    .login img:last-child {
+        display: none;
+    }
+}
```
Comment (YuKuroA) [inline task-board/src/components/header/header.css]: 1. Added line at the end of the file

File: task-board/src/components/signIn/signIn.css [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+    font-weight: 400;
+    font-size: 12px;
+    line-height: 15px;
+    color: #FFFFFF;
+    border: none;
+}
+
+.error-message {
+    color: red;
+}
```
Comment (YuKuroA) [inline task-board/src/components/signIn/signIn.css]: Added line at the end of the file

File: task-board/src/components/signIn/signIn.tsx (hunk 11 above)
Comment (YuKuroA) [reply task-board/src/components/signIn/signIn.tsx]: 1.  Moved call of the function `passLogin(login)` . Now it calls after user passed validation.

File: task-board/src/components/signIn/signIn.tsx (hunk 12 above)
Comment (YuKuroA) [reply task-board/src/components/signIn/signIn.tsx]: 1. Replaced operator "==" with "===" in this file. (I used operator "==" only in this file of project, so no need to replace it elsewhere)
2. Made correct spacing for operator "==="

File: task-board/src/components/signIn/signIn.tsx [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+                    <button type="submit">LOGIN</button>
+                </div>
+            </form>
+        </div>
+    )
+
+    return signIn
+}
+
+export default SignIn
```
Comment (YuKuroA) [inline task-board/src/components/signIn/signIn.tsx]: 1. Added line at the end of file.
2. Added seni-colon for lines: 6, 9, 35, 36, 39 60, 62, 65

File: task-board/src/components/taskBoard/taskBoard.css [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+        flex-direction: row;
+        align-items: baseline;
+    }
+
+    .addColumn {
+        display: flex;
+        justify-content: start;
+        margin: 50px 50px auto 10px;
+    }
+
```
Comment (YuKuroA) [inline task-board/src/components/taskBoard/taskBoard.css]: 1. Removed extra whiteline
2. Added line at the end of the file

File: task-board/src/components/taskBoard/taskBoard.tsx (hunk 13 above)
Comment (YuKuroA) [reply task-board/src/components/taskBoard/taskBoard.tsx]: 1. Removed space after useState (line 14)

Comment (YuKuroA) [reply task-board/src/components/taskBoard/taskBoard.tsx]: 2. Added line at the end of the file
3. Added semi-colons for lines: 50, 52, 55
4. Removed semi-colons from lines: 19, 23

File: task-board/src/pages/signIn/signInPage.css [hunk 25]
```diff
@@ -0,0 +1,15 @@
...
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
```
Comment (YuKuroA) [inline task-board/src/pages/signIn/signInPage.css]: 1. Removed extra whiteline at the end of the file

File: task-board/src/pages/userBoard/userBoard.css [hunk 26]
```diff
@@ -1,8 +1,10 @@
 body {
   background-color: #29487C;
+
 }
```
Comment (YuKuroA) [inline task-board/src/pages/userBoard/userBoard.css]: 1. Removed extra whitelines
2. Added line at the end of file

File: task-board/src/pages/userBoard/userBoard.tsx (hunk 14 above)
Comment (YuKuroA) [reply task-board/src/pages/userBoard/userBoard.tsx]: 1. Removed extra white lines
2. Added semi-colons for lines: 11, 21, 23, 26
3. Added line at the end of the file

File: task-board/src/pages/userBoard/userBoard.tsx (hunk 15 above)
Comment (YuKuroA) [reply task-board/src/pages/userBoard/userBoard.tsx]: Resolved

File: task-board/src/pages/userBoard/userBoard.tsx (hunk 18 above)
Comment (YuKuroA) [reply task-board/src/pages/userBoard/userBoard.tsx]: Resolved

File: task-board/src/pages/userBoard/userBoard.tsx (hunk 16 above)
Comment (YuKuroA) [reply task-board/src/pages/userBoard/userBoard.tsx]: Resolved

File: task-board/src/pages/userBoard/userBoard.tsx (hunk 17 above)
Comment (YuKuroA) [reply task-board/src/pages/userBoard/userBoard.tsx]: Resolved

File: task-board/src/components/card/card.tsx [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+            cardName: target.cardName.value
+        });
+
+        setShowEditCardForm(false);        
+    }
+    
+    const Card = (
+        <div>
+            {
+            !showEditCardForm &&
```
Comment (IlIAntonIlI) [inline task-board/src/components/card/card.tsx]: tab missed

File: task-board/src/components/column/column.tsx [hunk 28]
```diff
@@ -85,15 +85,21 @@ const Colunm: React.FC<ColumnProps> = ( props ) => {
                     )}                   
                 </Droppable>
                 <div className="addCard">
-                    {!showForm && <label onClick={openModal}>+ Add new card</label>}
-                    {showForm && <AddCard addCard={addCard} closeModal={closeModal} />}
+                    {
+                    !showForm && 
+                        <label onClick={openModal}>+ Add new card</label>
+                    }
```
Comment (IlIAntonIlI) [inline task-board/src/components/column/column.tsx]: Missed one tab, should be:
```
{
  !showForm && 
    <label onClick={openModal}>+ Add new card</label>
}
```
or

```
{
  !showForm && <label onClick={openModal}>+ Add new card</label>
}
```
both are ok

**Comment by IlIAntonIlI** [comment]
Almost done 👍 
Also you can just respond with `fixed` or smth like that in threads not to lose time to search places where changes was done

File: task-board/src/components/card/card.tsx (hunk 27 above)
Comment (YuKuroA) [reply task-board/src/components/card/card.tsx]: Resolved

File: task-board/src/components/column/column.tsx (hunk 28 above)
Comment (YuKuroA) [reply task-board/src/components/column/column.tsx]: Resolved

Comment (IlIAntonIlI) [reply task-board/src/components/column/column.tsx]: its would be better to add one tab before  <label onClick={openModal}>+ Add new card</label> as well
But, anyways, looks good at the moment

**Comment by IlIAntonIlI** [comment]
Approved

**Comment by IlIAntonIlI** [comment]
You can merge your changes

**Comment by IlIAntonIlI** [comment]
Deploy to vercel

**Comment by IlIAntonIlI** [comment]
Pascal case in the names of components

File: task-board/src/components/signIn/signIn.tsx
Comment (IlIAntonIlI) [inline task-board/src/components/signIn/signIn.tsx:1]: rename signIn to Auth and use react router to display `/sign-in`, `/sign-up` etc

File: task-board/src/App.tsx [hunk 29]
```diff
@@ -0,0 +1,26 @@
...
+  const [userLogin, setUserLogin] = useState('');
+
+  const passLogin = (login:string) => {
+    setUserLogin(login);
+  };
+  
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path='*' element={<SignInPage passLogin={passLogin} />} />
```
Comment (IlIAntonIlI) [inline task-board/src/App.tsx]: place pattern route `*` always at the end, otherwise it will display signIn page for any route

**Pull request merged**